src/every.test.ts

```diff
@@ -0,0 +1,81 @@
+import { every } from "./every";
+import { isNumber } from "./isNumber";
+import { pipe } from "./pipe";
+
+describe("data-first", () => {
+  test("returns correct values on simple arrays", () => {
+    const trueInput = [1, 2, 3];
+    const trueData = every(trueInput, (val) => val < 5);
+    expect(trueData).toBe(true);
+
+    const falseInput = ["foo", "bar", "buzz"];
+    const falseData = every(falseInput, (val) => val.length > 3);
+    expect(falseData).toBe(false);
+  });
+
+  // empty set check
+  test("returns true on empty arrays regardless of predicate", () => {
+    const input = [] as const;
+    const data = every(input, () => false);
+    expect(data).toBe(true);
+  });
+
+  test("narrows types when a type predicate is passed", () => {
+    const input: Array<number | string> = [1, 2];
+    const data = every(input, isNumber);
+    expect(data).toBe(true);
+    if (every(input, isNumber)) {
+      expectTypeOf(input).toEqualTypeOf<Array<number>>();
+    }
+  });
+});
+
+describe("data-last", () => {
+  test("returns correct values on simple arrays", () => {
+    const trueInput = [1, 2, 3];
+    const trueData = every((val: number) => val < 5)(trueInput);
+    expect(trueData).toBe(true);
+
+    const falseInput = ["foo", "bar", "buzz"];
+    const falseData = every((val: string) => val.length > 3)(falseInput);
+    expect(falseData).toBe(false);
+  });
+
+  // empty set check
+  test("returns true on empty arrays regardless of predicate", () => {
+    const input = [] as const;
+    const data = every(() => false)(input);
+    expect(data).toBe(true);
+  });
+
+  test("narrows types when a type predicate is passed", () => {
+    const input: Array<number | string> = [1, 2];
+    const data = every(isNumber)(input);
+    expect(data).toBe(true);
+    if (every(isNumber)(input)) {
+      expectTypeOf(input).toEqualTypeOf<Array<number>>();
```

Comment:
don't mix type testing with runtime testing; they run at different times in the test cycle and thus bind two unrelated concerns.

That's why I prefer having a "runtime" block and a "typing" block.